src/main/scala/com/microsoft/hyperspace/index/FileBasedSignatureProvider.scala

```diff
@@ -55,7 +55,7 @@ class FileBasedSignatureProvider extends LogicalPlanSignatureProvider {
           _,
           _,
           _) =>
-        fingerprint ++= location.allFiles.foldLeft("")(
+        fingerprint ++= location.allFiles.sortBy(_.hashCode).foldLeft("")(
```

Comment:
is `hashCode` enough? What if two strings return the same hashCode?

Reply:
Maybe use `sortWith` to compare first with `hashCode` then falls back to string comparison (very rare)?

Reply:
hash code function:
``` 
public int hashCode() {
        if (hash != 0)
            return hash;
        int h = hashIgnoringCase(0, scheme);
        h = hash(h, fragment);
        if (isOpaque()) {
            h = hash(h, schemeSpecificPart);
        } else {
            h = hash(h, path);
            h = hash(h, query);
            if (host != null) {
                h = hash(h, userInfo);
                h = hashIgnoringCase(h, host);
                h += 1949 * port;
            } else {
                h = hash(h, authority);
            }
        }
        hash = h;
        return h;
    }```

how about using XOR or just compare all file list like hybrid scan?

Reply:
btw changing signature is a breaking change.... 😢

Reply:
> btw changing signature is a breaking change.... 😢

I think we can maintain the backward compatibility if we want by creating `IndexSignatureProviderV2`: https://github.com/microsoft/hyperspace/blob/199fa9a0ee609ec498d6ebfbab2d44e8bd11fd9c/src/main/scala/com/microsoft/hyperspace/index/LogicalPlanSignatureProvider.scala#L55

Reply:
> compare all file list like hybrid scan?

What do you mean by this?

Reply:
> What do you mean by this?

means not using FileBasedSignature and check all files (name/len/modification time) ? sorting + hash computation is also  not cheap.

Reply:
> sorting + hash computation is also not cheap.

Are we looking at 10s, 100s or 1000s overhead?

> means not using FileBasedSignature and check all files (name/len/modification time) ?

Signature was meant to quickly rule out indexes looking at source files, but it may not be useful with hybrid scan anymore.  We can address this as a follow up PR, but let's fix this first with sorting (maybe `sortBy(_.getPath.toString)`?) since I see tests are failing due to this.

Reply:
While testing 100k chunk dataset, I observed the overhead is similar to Hybrid Scan in case there's no candidate index (partial index case). A tag for signature value might be helpful to reduce the overhead.

Reply:
> I observed the overhead is similar to Hybrid Scan in case there's no candidate index (partial index case)

Is this good, bad, or reasonable?